File: Javascript/Arrays & Hashing/Valid-Anagram.js
/**
 *
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */

const isAnagram = (s, t) => {
  if (s.length != t.length) return false;

  const frequency = new Array(26).fill(0);

  for (let i = 0; i < s.length; i++) {
    const [sChar, tChar] = [s[i].charCodeAt(0), t[i].charCodeAt(0)];
    const a = "a".charCodeAt(0);

    frequency[sChar - a]++;
    frequency[tChar - a]--;
  }

  return frequency.every((freq) => freq == 0);
};

File: Rust/Arrays & Hashing/Valid-Anagram.rs
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.len() != t.len() {
            return false;
        }

        let mut count = [0; 26];

        for (s_char, t_char) in s.chars().zip(t.chars()) {
            count[(s_char as u8 - b'a') as usize] += 1;
            count[(t_char as u8 - b'a') as usize] -= 1;
        }

        count.iter().all(|x| *x == 0)
    }
}